db/sql/05_msar.sql

```diff
@@ -4945,8 +4942,11 @@ BEGIN
             WHERE contype = 'f' AND conrelid = contextual_tab_id AND conkey = array[fk_col_id];
 
             IF ref_tab_id IS NULL THEN
-              -- Silently ignore references to non-FK columns. This can happen if the constraint
-              -- has been dropped.
+              CONTINUE template_parts_loop;
+            END IF;
+
+            ref_col_name := msar.get_column_name(ref_tab_id, ref_col_id);
+            IF ref_col_name IS NULL THEN
               CONTINUE template_parts_loop;
             END IF;
```

Comment:
Since `fk_col` is dependent on `ref_col` via a foreign key constraint, it is not possible to drop `ref_col`:
- Without dropping the foreign key constraint. _or_
- Without dropping the dependent cols, `fk_col` in this case.

So, this check is redundant as we will never encounter this scenario. Please revert this.